### ISSUE TITLE
Upcoming changes to Tweets

### ISSUE THREAD

**Issue by JoeMayo** [JoeMayo/LinqToTwitter#61]
Several fields on Status changes:

https://dev.twitter.com/overview/api/upcoming-changes-to-tweets

**Comment by masonkmeyer** [comment]
@JoeMayo We are very interested in using the new Extended Tweet.  Are you planning on supporting both the extended and classic tweet format? Also, I didn't see an actual release date. Any ideas when they changes are effective?

**Comment by JoeMayo** [comment]
Hi Mason,

Yes, I believe Twitter has a fallback strategy and LINQ to Twitter will support that. Though timing varies, I can see growing interest in this feature.

[@JoeMayo](https://twitter.com/JoeMayo)

**Comment by Kew** [comment]
I'm interested in this please.

**Comment by DrMiaow** [comment]
+1

Apparently these changes take effect on the 19th Sept.

http://www.theverge.com/2016/9/12/12891562/twitter-tweets-140-characters-expand-photos

Will this 'just work' or do we need to change the way we need to call the API and depend upon an update in LinqToTwitter? It's not completely obvious from the link. https://dev.twitter.com/overview/api/upcoming-changes-to-tweets It looks like we won't need to do anything unless we want to display tweets, in which case we need to request the tweets in an extended mode?

**Comment by DrMiaow** [comment]
It's happened! Will test tonight if it works without an API change. 

https://twitter.com/i/web/status/777915304261193728

**Comment by Kew** [comment]
Looks like it works without a change needed :) https://twitter.com/qewbe/status/779913836115562496

![image](https://cloud.githubusercontent.com/assets/1836917/18813181/255a6288-8334-11e6-935a-1c1b96e2ca3d.png)

**Comment by bluejusurf** [comment]
This doesn't seem to work for me, I'm fairly new to Linq to Twitter so not sure if I'm missing something, if I make a call

            var tweets = (from t in twitterContext.Status
                          where t.Type == StatusType.User
                          && t.ScreenName == searchString
                          && t.Count == numItems
                          select t);

then any short tweets with an image are returned with the image included in the media entities, but longer tweets that also have an image are returned with media entities set to null.

I have done a bit of testing in a separate project making a call to 
https://api.twitter.com/1.1/statuses/user_timeline.json?screen_name={0}&include_rts=1&exclude_replies=1&count=5&tweet_mode=extended

with the screen_name parameter set to my screen name returns the media entities in all tweets, if I exclude the "tweet_mode=extended" from the url I get exactly the same as I get from the LinqToTwitter calls.

**Comment by mtcomputergirl** [comment]
Wondering status on this? I need this functionality as well.

**Comment by JoeMayo** [comment]
I just pushed changes for this and deployed a new beta to NuGet at https://www.nuget.org/packages/linqtotwitter/4.2.0-beta2

There are a lot of moving parts here, but I added support for the new parameters to ReplyAsync and TweetAsync. I also added a TweetMode parameter to all Status, Search, and Streaming queries.

Let me know if you have any problems.

**Comment by sadullahceran** [comment]
I have downloaded 4.2.0-beta2, and t seems it works flawlessly

**Comment by zaiboot** [comment]
Seems to be working fine in my local environment.

**Comment by rttremaine** [comment]
I'm using 4.2.0-beta2 in a .Net Core 2.0 project and the TweetMode property does not seem to exist, and so I can only get back the standard tweets (some of which are being truncated).

e.g. the commented-out line is needed (should be working?) but currently isn't...

```
List<Status> tweets = await ctx.Status.Where(t =>
  t.Type == StatusType.User &&
//  t.TweetMode == TweetMode.Extended &&  // TweetMode not available
  t.ScreenName == name
).ToListAsync();
```

I look forward to a fix when it becomes available.

**Comment by flyinmryan** [comment]
I was using version 4.1, but after seeing this post I downloaded 4.2.0-beta2.  I cannot seem to get any Media URLs for the images associated with Tweets.  I am getting the List<Status> back with 200 tweets, but the "MediaEntities" and "UrlEntities" within "ExtendedEntities" doesn't contain any information.  Is this a known issue?

**Comment by JoeMayo** [comment]
This was released in [LINQ to Twitter v4.2.0](https://github.com/JoeMayo/LinqToTwitter/releases/tag/v4.2.0).

[@JoeMayo](https://twitter.com/JoeMayo)
